Bukkit/src/main/java/com/plotsquared/bukkit/listener/PaperListener.java

```diff
@@ -429,18 +426,22 @@ public void onBeaconEffect(final BeaconEffectEvent event) {
         PlotPlayer<Player> plotPlayer = BukkitUtil.adapt(player);
         Plot playerStandingPlot = playerLocation.getPlot();
         if (playerStandingPlot == null) {
-            if (area.getRoadFlag(DisableBeaconEffectOverflowFlag.class)
-                    || Settings.Paper_Components.DISABLE_BEACON_EFFECT_OVERFLOW) {
+            if (!area.getRoadFlag(BeaconEffectsFlag.class) ||
+                    (beaconPlot != null && Settings.Paper_Components.DISABLE_BEACON_EFFECT_OVERFLOW)) {
                 event.setCancelled(true);
             }
             return;
         }
 
-        if (beaconPlot.equals(playerStandingPlot)) {
+        boolean plotBeaconEffects = playerStandingPlot.getFlag(BeaconEffectsFlag.class);
```

Comment:
This would NPE on spigot platforms as we do not set the flag to the global container. You should use `playerStandingPlot.getFlagContainer().getFlag()` and then check for nullability before doing `Flag#getValue`. The same issue will occur for getting the road flag.

OR

We could still use the flag for spigot, where any beacon effect is cancelled based on the player's location, and add a new listener that's only enabled if paper is not present. Perhaps in a new class called "SpigotListener". @IntellectualSites/plotsquared-team thoughts on this?

Reply:
I agree. Having all registrations in one place also makes the code a lot clearer.